prioritization_fees/src/block_priofees.rs

```diff
@@ -29,13 +26,33 @@ pub struct PrioFeesService {
 impl PrioFeesService {
     pub async fn get_latest_priofees(&self) -> Option<(Slot, PrioFeesStats)> {
         let lock = self.block_fees_store.recent.read().await;
-        let latest_in_store = lock.last_key_value();
-        latest_in_store.map(|x| (*x.0, x.1.clone()))
+        lock.last_key_value()
+            .map(|(slot, value)| (*slot, value.calculate_stats()))
+    }
+
+    pub async fn get_last_n_priofees_aggregate(&self, nb: usize) -> Option<(Slot, PrioFeesStats)> {
+        let lock = self.block_fees_store.recent.read().await;
+        let last_slot = match lock.last_key_value().map(|x| *x.0) {
+            Some(slot) => slot,
+            None => {
+                return None;
```

Comment:
maybe you should also abort if there are less than `nb` items in the map?

Reply:
will do it later